app/controllers/sites_controller.rb

```diff
@@ -200,9 +204,20 @@ def nav_menu
 
   def get_project
     @project = Project.find(params[:project_id])
-
     # avoid the same project assigned more than once to a site
     @site.projects << @project if defined?(@site) && !@site.projects.include?(@project)
+    @permissions = Access::ByPermission.sites(current_user)
+  end
+
+  def get_project_if_exists
+    @project = (Project.find(params[:project_id]) if params.key?(:project_id))
+    # avoid the same project assigned more than once to a site
+    @site.projects << @project if defined?(@site) && defined?(@project) && !@site.projects.include?(@project)
+    @permissions = if @project.nil?
+                     Access::ByPermission.sites(current_user)
+                   else
+                     Access::ByPermission.sites(current_user, project_ids: [@project.id])
+                   end
```

Comment:
split this out into a different function and don't assign the result to an instance variable - just return it

named something like `list_permissions`